pkg/fqdn/cache.go

```diff
@@ -108,15 +108,39 @@ type DNSCache struct {
 	// This map is subordinate to forward, above. An IP inserted into forward, or
 	// expired in forward, should also be added/removed in reverse.
 	reverse map[string]nameEntries
+
+	// LastCleanup is the last time that the cleanup happens.
+	lastCleanup time.Time
+
+	// cleanup is the map of TTL expiration time and the entries that are going
+	// to expire will be deleted by CleanupExpiredEntries.
+	cleanup map[int64][]string
+
+	// overLimit is a map with the host names that has reached the ips limits
+	// defined by the limit field
+	overLimit map[string]bool
+
+	// perHostLimit is the number of maximum number of IP per host.
+	perHostLimit int
 }
 
 // NewDNSCache returns an initialized DNSCache
 func NewDNSCache() *DNSCache {
 	c := &DNSCache{
-		forward: make(map[string]ipEntries),
-		reverse: make(map[string]nameEntries),
+		forward:     make(map[string]ipEntries),
+		reverse:     make(map[string]nameEntries),
+		lastCleanup: time.Now(),
+		cleanup:     map[int64][]string{},
+		overLimit:   map[string]bool{},
```

Comment:
since we set every other field, perhaps we should also set perHostLimit. It is redundant but makes it clear that it was intended. shrug.